### PR TITLE
[12.0][FIX] event_sale: properly compute if event is registrable

### PR DESCRIPTION
**Pull request by yajo**
The current algorithm was ignoring if seats were available and if tickets were expired.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
@Tecnativa TT17632

### CODE REVIEW

**Comment by Yajo** [comment]
Any human could review this please? :roll_eyes:

**Comment by pedrobaeza** [comment]
@tde-banana-odoo are you the responsible of this as the suggestion indicates?

**Comment by tde-banana-odoo** [comment]
Event is in marketing, indeed... as far as I remember templates correctly manage seats available, am I wrong ? I know the is_registrable is a bit buggy, but if templates are correct...

**Comment by Yajo** [comment]
I see... :smile: You guys at Odoo sometimes seem to love your bugs. :heart: :bug: 

OK, 1st thing is that this code is in `event_sale` and website templates are in `website_event_sale`, so this method should be reliable no matter the context. If its purpose is not to know whether an event is registrable, then start by not calling it `_is_event_registrable()`. :grin: 

Regarding your concern, with an event like this:



Right now it looks like this:




With the patch, it would look like this:

File: addons/event_sale/models/event.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def _is_event_registrable(self):
         self.ensure_one()
         if not self.event_ticket_ids:
             return True
-        return all(self.event_ticket_ids.with_context(active_test=False).mapped(lambda t: t.product_id.active))
+        return bool(self.event_ticket_ids.with_context(active_test=False).filtered(lambda t: t.product_id.active and not t.is_expired and t.seats_available))
```
Comment (chienandalu) [inline addons/event_sale/models/event.py:62]: You should consider the case when `seats_max` is 0, wich for Odoo mean unlimited seats

Comment (chienandalu) [reply addons/event_sale/models/event.py:62]: Maybe something like:
```suggestion
        return bool(self.event_ticket_ids.with_context(active_test=False).filtered(lambda t: t.product_id.active and not t.is_expired and t.seats_available if t.seats_max else True))
```

**Comment by Yajo** [comment]
Rebased and updated according to review.

**Comment by pedrobaeza** [comment]
@tde-banana-odoo what do you think now about this?

**Comment by tde-banana-odoo** [comment]
I will re-test in 12.0 to check the fix.

**Comment by tde-banana-odoo** [comment]
This is in my todo-list. As a side note we are quite busy on a big event update for 13.3, will have more time soon to get back at this one (hopefully).

**Comment by pedrobaeza** [comment]
Thanks for the follow-up!

**Pull request closed without merging**